### DISCUSSION TITLE
Problem with multiple REST API requests at once

### DISCUSSION THREAD

**Discussion by sinavali** [co-cart/co-cart#385] [Q&A] [answered]
i have a if statement in my code that will filter some cart items and after that it will create for loop to get all item_key(s) and send them to remove item api.
the problem is, when requests have been sent and resolved the responses, everything in the responses are good, but after that when i send GetCart request, the products (mines the last one) are still in the cart. so for 3 remove_from_cart req, only one (last) will remove...
this problem can be resolved if there was an api to bulk remove items from cart.

thanks in advance

**Comment by seb86** [comment, marked as answer]
@sinavali Sounds like the session didn't save changes made in time before the next request to remove another item was processed. Bulk requests is coming in the next big update if feedback on the beta goes well.

**Reply by sinavali** [reply]
is it implemented in beta version? and when next major update will release...?

**Reply by seb86** [reply]
I'm currently preparing to have a beta release out soon with the hopes to have it out officially mid next month if all community tests go well.
